pkgs/getzones/getzones.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2086
+
+set -eu
+set -o pipefail
+
+CAASCAD_ZONES_URL=${CAASCAD_ZONES_URL:-https://git.corp.cloudwatt.com/caascad/caascad-zones/raw/master/zones.json}
+CONFIG_DIR="$HOME/.config/kswitch"
+CAASCAD_ZONES_FILE="$CONFIG_DIR/zones.json"
+KSWITCH_DEBUG=${KSWITCH_DEBUG:-0}
+INFRA_ZONES_NAMES=(
+  infra-stg
+  infra-prd
+)
+
+log() {
+  echo -e "\x1B[32m--- $*\x1B[0m" >&2
+}
+
+log_debug() {
+  if [ $KSWITCH_DEBUG -eq 1 ]; then
+    echo -e "\x1B[34m--- $*\x1B[0m" >&2
+  fi
+}
+
+log-error() {
+  echo -e "\x1B[31m--- $*\x1B[0m" >&2
+}
+
+run() {
+  echo -e "\x1B[33m+++ $1\x1B[0m" >&2
+  eval "$1"
+}
+
+run_c() {
+  echo -e "\x1B[33m+++ $1\x1B[0m" >&2
+  _continue
+  eval "$1"
+}
+
+_continue() {
+  read -p "Continue [y/n]: " -n 1 -r
+  echo
+  [[ $REPLY =~ ^[Yy]$ ]] || exit 1
+}
+
+bash_completions() {
+  cat <<EOF
+_getzones_completions() {
+  local curr_arg;
+  curr_arg=\${COMP_WORDS[COMP_CWORD]}
+  if [ \${#COMP_WORDS[@]} -ge 3 ]; then
+    return
+  fi
+}
+complete -F _getzones_completions getzones
+EOF
+}
+
+usage() {
+  cat <<EOF
+Usage:
+  getzones PARENT_ZONE_NAME     Get a zones list of OCB zones and client zones attached to an INFRA_ZONE_NAME
+  getzones -h, --help           This help
+  getzones -v, --version        Current getzones version
+
+getzones automatically give a list of OCB and Client zones attached to an INFRA_ZONE_NAME.
```

Comment:
To be useful I think it should handle the most common case: find child zones of some parent zone

Reply:
It is not the same, but with a better explanation ?